parlai/core/agents.py

```diff
@@ -388,6 +388,10 @@ def load_agent_module(opt):
     optfile = model_file + '.opt'
     if os.path.isfile(optfile):
         new_opt = _load_opt_file(optfile)
+        # TODO we need a better way to say these options are never copied...
+        if 'datapath' in new_opt:
+            # never use the datapath from an opt dump
+            del new_opt['datapath']
```

Comment:
i'm not sure this is what we want to do. we should probably do something like `new_opt['datapath'] = opt['datapath']`, right? if you check line 446, you'll see that the model is loaded with `new_opt`

Reply:
The case you're concerned about is handled in lines: 406-408

https://github.com/facebookresearch/ParlAI/blob/cc8dfa4e4becc528f16e1f44e23fb97ed726c929/parlai/core/agents.py#L406-L408

Reply:
ah thanks